### DISCUSSION TITLE
Slow page load due to repeater

### DISCUSSION THREAD

**Discussion by aconjaer** [Budibase/budibase#14244] [Help]
I'm facing performace issues as the page load of my app takes more or less 7 seconds.
The main issue seems to be a repeater, which is only repeating 12 lines of a table. However, there are a lot of containers, labels and buttons nested under the repeater, 113 in total. Most of them are containers to organize the labels and buttons though. Are there some best practices regarding repeaters? Am I demanding too much from the app?
I'm self hosting budibase (version 2.29.18) and I don't see any CPU or memory issues on the server.

![image](https://github.com/user-attachments/assets/84ba6b99-3c7a-4582-938e-530f40202017)

**Comment by Hades1280** [comment]
you can put limit to 1 in Data Provider Settings, before you repeater. Default is 50 i think. 
![Captura de pantalla 2024-08-08 153456](https://github.com/user-attachments/assets/c4371f82-19b1-4035-b6bb-3dbbbaf350a3)

i hope this help to you.
